src/Controls/src/Core/Platform/ModalNavigationManager/ModalNavigationManager.Android.cs

```diff
@@ -398,6 +399,54 @@ public CustomComponentDialog(Context context, int themeResId) : base(context, th
 					this.OnBackPressedDispatcher.AddCallback(new CallBack(true, this));
 				}
 
+				public override bool OnKeyLongPress([GeneratedEnum] Keycode keyCode, KeyEvent e)
+				{
+					var activity = Context?.GetActivity();
+
+					if (activity is null)
+					{
+						return false;
+					}
+
+					return activity.OnKeyLongPress(keyCode, e);
+				}
+
+				public override bool OnKeyDown([GeneratedEnum] Keycode keyCode, KeyEvent e)
```

Comment:
I worry about firing through to the activity methods because that is going to possibly trigger other side effects

Here's the source code on onKeyUp from android

```java
    @Override
    public boolean onKeyDown(int keyCode, KeyEvent event) {
        if (performMenuItemShortcut(event)) {
            return true;
        }
        return super.onKeyDown(keyCode, event);
    }
```

I'm not too sure the best path to take here

It seems like the keylistener and keyevent doesn't really express enough to be useful

Long term I think we could pipe these through platform lifecycle events but I know that doesn't super help right now in net9